File: frontend/src/types/detection.ts
/** 后端 API 响应外壳 */
export interface DetectionApiResponse<T> {
  x_request_id: string
  msg: string
  data: T
  code: number
}

/** 检测接口 data 结构 */
export interface DetectionApiData {
  result: DetectionResult
  file_type: string
  file_data: string
}

/** image_property.ps 单算法结果 */
export interface PsResult {
  tampered_positions: number[][]
  is_tampered: boolean
  image?: string
  tampered_scores: number[][]
}

/** image_property（可扩展其他算法） */
export interface ImageProperty {
  ps: PsResult
  [key: string]: unknown
}

/** 篡改检测结果（即后端 data.result） */
export interface DetectionResult {
  image_width: number
  image_height: number
  image_property: ImageProperty
}

/** 历史记录项（图片列表） */
export interface HistoryItem {
  id: number
  name: string
  file?: File
  preview: string
  result?: DetectionResult
}

File: backend/app/schemas/detection.py
"""Request/Response schemas for tamper detection API."""
from pydantic import BaseModel, Field


class PSResult(BaseModel):
    """PS检测结果"""
    tampered_positions: list[list[int]] = Field(default_factory=list, description="篡改位置坐标")
    is_tampered: bool = Field(..., description="是否存在篡改")
    image: str = Field(default="", description="图像数据")
    tampered_scores: list[list[float]] = Field(default_factory=list, description="篡改置信度分数")


class ImageProperty(BaseModel):
    """图像属性"""
    ps: PSResult = Field(..., description="PS检测结果")


class DetectionResult(BaseModel):
    """检测结果"""
    image_width: int = Field(..., description="图像宽度")
    image_property: ImageProperty = Field(..., description="图像属性")
    image_height: int = Field(..., description="图像高度")


class Data(BaseModel):
    """响应数据"""
    result: DetectionResult = Field(..., description="检测结果")
    file_type: str = Field(default="", description="文件类型")
    file_data: str = Field(default="", description="文件数据")


class DetectionResponse(BaseModel):
    """Tamper detection API response."""
    x_request_id: str = Field(..., description="请求ID")
    msg: str = Field(default="success", description="响应消息")
    data: Data = Field(..., description="响应数据")
    code: int = Field(default=200, description="响应代码")


class HealthResponse(BaseModel):
    """Health check response."""
    status: str = "ok"
    service: str = "tamper-detection"
